JSON/fio-multi-params-test.json

```
@@ -1,67 +1,63 @@
 {
     "global-options": [
-	{
-	    "name": "common-params",
-	    "params": [
-		{ "arg": "bs", "vals": [ "4k", "8k" ] },
-		{ "arg": "norandommap", "vals":  [ "ON" ] },
-		{ "arg": "time_based", "vals": [ "1" ] },
-		{ "arg": "runtime", "vals": [ "120" ] },
-		{ "arg": "ramp_time", "vals": [ "5" ] },
-		{ "arg": "size", "vals": [ "10g" ] },
-		{ "arg": "clocksource", "vals": [ "gettimeofday" ] },
-		{ "arg": "filename", "vals": [ "/dev/foobar" ] },
-		{ "arg": "direct", "vals": [ "1" ] },
-		{ "arg": "sync", "vals": [ "0" ] },
-		{ "arg": "jobfile", "vals": [ "EMPTY_JOB_FILE" ] }
-	    ]
-	},
-	{
-	    "name": "crucible-defaults",
-	    "params": [
-		{ "arg": "write_hist_log", "vals": ["fio"] },
-		{ "arg": "log_hist_msec", "vals": ["10000"] },
-		{ "arg": "write_bw_log", "vals": ["fio"] },
-		{ "arg": "write_iops_log", "vals": ["fio"] },
-		{ "arg": "write_lat_log", "vals": ["fio"] },
-		{ "arg": "log_avg_msec", "vals": ["1000"] },
-		{ "arg": "log_unix_epoch", "vals": ["1"] },
-		{ "arg": "output-format", "vals": ["json"] },
-		{ "arg": "output", "vals": ["fio-result.json"] }
-	    ]
-	},
-	{
-	    "name": "jobs",
-	    "params": [
-		{ "arg": "name", "vals": [ "job1" ] }
-	    ]
-	}
+    {
+        "name": "common-params",
+        "params": [
+        { "arg": "bs", "vals": [ "4k", "8k" ] },
+        { "arg": "norandommap", "vals":  [ "ON" ] },
+        { "arg": "time_based", "vals": [ "1" ] },
+        { "arg": "runtime", "vals": [ "120" ] },
+        { "arg": "ramp_time", "vals": [ "5" ] },
+        { "arg": "size", "vals": [ "10g" ] },
+        { "arg": "clocksource", "vals": [ "gettimeofday" ] },
+        { "arg": "filename", "vals": [ "/dev/foobar" ] },
+        { "arg": "direct", "vals": [ "1" ] },
+        { "arg": "sync", "vals": [ "0" ] },
+        { "arg": "jobfile", "vals": [ "EMPTY_JOB_FILE" ] }
```

This is definitely me being very pedantic, but this type of indentation really bugs me because I think its hard to read.  Should everything in the "params" array be indented another level?

---

> This is definitely me being very pedantic, but this type of indentation really bugs me because I think its hard to read. Should everything in the "params" array be indented another level?

ok, so to be consistent, I think we should do this:
```suggestion
  5         "params": [
  6         {
  7             "arg": "bs", "vals": [ "4k", "8k" ], "role": "client"
  8         },
`````
That is, whenever a block starts with { or [, it is added to the end of the line, and the next level starts in the same indentation level as the line above. In this example: params is array and has [ at the end, and in the next line { goes to the same indentation as params.

---

so I was thinking more along these lines: